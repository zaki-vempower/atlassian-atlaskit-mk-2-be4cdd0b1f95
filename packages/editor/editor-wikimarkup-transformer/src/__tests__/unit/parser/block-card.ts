import WikiMarkupTransformer from '../../../index';

describe('JIRA wiki markup - Block Card', () => {
  const testCases: Array<[string, string]> = [
    [
      'should create Block Card between lines of text',
      `line 1 
      [http://...|http://...|smart-card]
      line 2 `,
    ],
    [
      'should create Block Card for old representation between lines of text',
      `line 1 
      [http://...|http://...|block-link]
      line 2 `,
    ],
    [
      'should create Block Card with nothing else in the document',
      `[http://...|http://...|smart-card]`,
    ],
    [
      'should create Block Card with whitespace on the same line',
      ` [http://...|http://...|smart-card] 
      [http://...|http://...|smart-card]`,
    ],
    [
      'should create Block Card if preceded and followed by text', // This will result in the text being moved to another line
      `
      abc [http://...|http://...|smart-card] def
      `,
    ],
    [
      'should create Block Cards in table',
      `|[http://...|http://...|smart-card]|`,
    ],
    [
      'should create Block Cards in panel', // Block ends up outside of the panel in adf
      `|{panel:bgColor=#deebff}test [http://...|http://...|smart-card] test{panel}|`,
    ],
    [
      'should recover existing Block Cards',
      `|{adf:display=block}
      {"type":"blockCard","attrs":{"url":"https://aolrich-automation.atlassian.net/browse/SOF-1"}}
      {adf}|{adf:display=block}
      {"type":"blockCard","attrs":{"url":"https://aolrich-automation.atlassian.net/browse/SOF-1"}}
      {adf}|
      [https://aolrich-automation.atlassian.net/browse/SOF-1|https://aolrich-automation.atlassian.net/browse/SOF-1|smart-card]`,
    ],
  ];

  for (const [testCaseDescription, markup] of testCases) {
    it(testCaseDescription, () => {
      const transformer = new WikiMarkupTransformer();
      expect(transformer.parse(markup)).toMatchSnapshot();
    });
  }
});
