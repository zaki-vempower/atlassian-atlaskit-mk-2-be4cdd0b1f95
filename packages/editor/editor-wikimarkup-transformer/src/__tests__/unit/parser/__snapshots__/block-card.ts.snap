// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JIRA wiki markup - Block Card should create Block Card between lines of text 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "line 1 ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "data": null,
        "url": "http://...",
      },
      "type": "blockCard",
    },
    Object {
      "content": Array [
        Object {
          "text": "line 2 ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Block Card should create Block Card for old representation between lines of text 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "line 1 ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "data": null,
        "url": "http://...",
      },
      "type": "blockCard",
    },
    Object {
      "content": Array [
        Object {
          "text": "line 2 ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Block Card should create Block Card if preceded and followed by text 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "abc ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "data": null,
        "url": "http://...",
      },
      "type": "blockCard",
    },
    Object {
      "content": Array [
        Object {
          "text": " def",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Block Card should create Block Card with nothing else in the document 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "data": null,
        "url": "http://...",
      },
      "type": "blockCard",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Block Card should create Block Card with whitespace on the same line 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "data": null,
        "url": "http://...",
      },
      "type": "blockCard",
    },
    Object {
      "content": Array [
        Object {
          "text": " ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "data": null,
        "url": "http://...",
      },
      "type": "blockCard",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Block Card should create Block Cards in panel 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "default",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "attrs": Object {
                    "panelType": "info",
                  },
                  "content": Array [
                    Object {
                      "content": Array [
                        Object {
                          "text": "test ",
                          "type": "text",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "type": "panel",
                },
                Object {
                  "attrs": Object {
                    "data": null,
                    "url": "http://...",
                  },
                  "type": "blockCard",
                },
                Object {
                  "attrs": Object {
                    "panelType": "info",
                  },
                  "content": Array [
                    Object {
                      "content": Array [
                        Object {
                          "text": " test",
                          "type": "text",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "type": "panel",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Block Card should create Block Cards in table 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "default",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "attrs": Object {
                    "data": null,
                    "url": "http://...",
                  },
                  "type": "blockCard",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Block Card should recover existing Block Cards 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "default",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "attrs": Object {
                    "data": null,
                    "url": "https://aolrich-automation.atlassian.net/browse/SOF-1",
                  },
                  "type": "blockCard",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "attrs": Object {
                    "data": null,
                    "url": "https://aolrich-automation.atlassian.net/browse/SOF-1",
                  },
                  "type": "blockCard",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
    Object {
      "content": Array [
        Object {
          "text": "      ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "data": null,
        "url": "https://aolrich-automation.atlassian.net/browse/SOF-1",
      },
      "type": "blockCard",
    },
  ],
  "type": "doc",
}
`;
